code/network/multi_pxo.cpp

```diff
@@ -520,21 +491,11 @@ int Multi_pxo_max_chat_display[GR_NUM_RESOLUTIONS] = {
 #define CHAT_MODE_CHANNEL_SWITCH		4			// "switching channels" message - draw in red
 #define CHAT_MODE_MOTD					5			// message of the day from the chat server
 
-typedef struct chat_line {
-	chat_line *next,*prev;
-	char text[MAX_CHAT_LINE_LEN+1];
-	int mode;
-} chat_line;
-
-// the chat linked list itself
-chat_line *Multi_pxo_chat = NULL;
-
-// the current add line
-chat_line *Multi_pxo_chat_add = NULL;
+// the chat list
+SCP_vector<chat_line> Multi_pxo_chat_vec;
```

Comment:
I'm not sure if it's the best way but I put in a quick limiter. After `MAX_CHAT_LINES` it will erase the first item in the vector before adding the new one. Seems to work ok with a few test runs.